### PR TITLE
Feature/dk 4

### PR DESCRIPTION
**Pull request by wansom**
I used a combination of stepper widget and animated switcher on the register page,haven't run the app to see how it appears on actual device

### CODE REVIEW

**Comment by CarltonK** [comment]
@wansom Please use single quotes. eslint is configured to use single quotes. Have you been running the firebase functions locally? Because the lint could catch those issues

**Comment by wansom** [comment]
I am used to writing my functions as express apps and testing them in
postman,am making the changes now

File: lib/screens/authentication/registration_screen.dart [hunk 1]
```diff
@@ -124,3 +59,490 @@ class RegistrationBody extends StatelessWidget {
...
+    );
+  }
+
+  /// Returns the next button.
+  Widget nextButton() {
+    return MaterialButton(
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(30),
+      ),
+      color: Colors.orange,
```
Comment (CarltonK) [inline lib/screens/authentication/registration_screen.dart:121]: @wansom Here you can call Theme.of(context).primaryColor

File: lib/screens/authentication/registration_screen.dart [hunk 2]
```diff
@@ -124,3 +59,490 @@ class RegistrationBody extends StatelessWidget {
...
+      padding: EdgeInsets.all(12),
```
Comment (CarltonK) [inline lib/screens/authentication/registration_screen.dart:122]: @wansom Add a const here and also to all padding or margin properties for performance optimization

File: lib/screens/authentication/registration_screen.dart [hunk 3]
```diff
@@ -124,3 +59,490 @@ class RegistrationBody extends StatelessWidget {
...
+                              padding: const EdgeInsets.all(6),
+                              child: TextField(
+                                decoration: InputDecoration(
+                                  labelText: 'sex',
+                                  prefixIcon: const Icon(Icons.email),
+                                ),
+                              ),
+                            ),
+                            const SizedBox(height: 16),
+                            Container(
```
Comment (CarltonK) [inline lib/screens/authentication/registration_screen.dart:295]: @wansom To save space, try and compact the first name and last name inside a row

File: lib/screens/authentication/registration_screen.dart [hunk 4]
```diff
@@ -124,3 +59,490 @@ class RegistrationBody extends StatelessWidget {
...
+                padding: const EdgeInsets.all(6),
+                child: TextField(
+                  decoration: InputDecoration(
+                    labelText: 'Password',
+                    prefixIcon: const Icon(Icons.vpn_key),
+                  ),
+                ),
+              ),
+              const SizedBox(height: 16),
+              Container(
```
Comment (CarltonK) [inline lib/screens/authentication/registration_screen.dart:533]: @wansom You can use a text editing controller to compare both values

File: lib/screens/authentication/registration_screen.dart [hunk 5]
```diff
@@ -124,3 +59,490 @@ class RegistrationBody extends StatelessWidget {
...
+                                decoration: InputDecoration(
+                                  labelText: 'Age',
+                                  prefixIcon: const Icon(Icons.person),
+                                ),
+                              ),
+                            ),
+                            const SizedBox(height: 16),
+                            Container(
+                              padding: const EdgeInsets.all(6),
+                              child: TextField(
```
[the remark this answers is not in the document]
Comment (wansom) [reply lib/screens/authentication/registration_screen.dart]: done

File: lib/screens/authentication/registration_screen.dart (hunk 2 above)
Comment (wansom) [reply lib/screens/authentication/registration_screen.dart]: done

**Pull request merged**